FinalProject/gs-rest-service-master/complete/src/main/java/dao/imp/CartDaoImp.java

```diff
@@ -0,0 +1,182 @@
+package dao.imp;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.transaction.annotation.Transactional;
+
+import bo.Cart;
+import bo.CartItem;
+import bo.Product;
+import bo.User;
+import connector.DbConnector;
+import dao.CartDao;
+
+public class CartDaoImp implements CartDao {
+
+	@Transactional
+	public int getLastId() throws SQLException {
+		Connection conn = DbConnector.getInstance().getConnection();
+		String query = "SELECT MAX(c.cartId) maxId FROM carts c";
+		ResultSet result;
+		int cartId = 0;
+
+		try {
+			PreparedStatement preparedStatement = conn.prepareStatement(query);
+			result = preparedStatement.executeQuery();
+			while (result.next()) {
+				cartId = result.getInt("maxId");
+			}
+		} catch (SQLException e) {
+			throw e;
+		}
+		return cartId;
+	}
+
+	@Transactional
+	public void addCart(Cart cart) throws SQLException {
+		Connection conn = DbConnector.getInstance().getConnection();
+		int cartId = cart.getCartId();
+		String userName = cart.getUser().getUserName();
+		String query = "INSERT INTO carts VALUES (?, ?)";
+		// String query2 = "INSERT INTO cartItems VALUES (?, ?, ?)";
+
+		try {
+			PreparedStatement preparedStatement = conn.prepareStatement(query);
+			preparedStatement.setInt(1, cartId);
+			preparedStatement.setString(2, userName);
+			preparedStatement.execute();
+			/*
+			 for (CartItem item : cart.getItems()) { preparedStatement =
+			 conn.prepareStatement(query2); preparedStatement.setInt(1,
+			 cartId); preparedStatement.setInt(2,
+			 item.getProduct().getProductId()); preparedStatement.setInt(3,
+			 item.getQuantity()); preparedStatement.execute(); }
+			 */
+		} catch (SQLException e) {
+			throw e;
+		}
+	}
+
+	@Transactional
+	public Cart getCart(int cartId) throws SQLException {
+		Connection conn = DbConnector.getInstance().getConnection();
+		String query = "SELECT * FROM carts c INNER JOIN users u ON c.userName = u.userName WHERE c.cartId = ?";
+		String query2 = "SELECT * FROM cartItems ci INNER JOIN products p ON ci.productId = p.productId WHERE ci.cartId = ?";
+		ResultSet result;
+		Cart cart = new Cart();
+
+		try {
+			PreparedStatement preparedStatement = conn.prepareStatement(query);

```

Comment:
Minor: You could avoid line 67 with  DbConnector.getInstance().getConnection().prepareStatement(query);